src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -209,12 +207,29 @@ public long getPower() {
     @Override
     public void update() {
 
-        //container.givePower(1000,false);
-
-
+        //checking for the machine type
         if (machineTier == null)
             machineTier = MachineTier.byMeta(getBlockMetadata());
 
+
+        if (container.getStoredPower() <= 0 && machineTier != MachineTier.TIER_0 && machineActive != false || fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && machineActive != false) {
+            machineActive = false;
+            this.markForUpdate();
+        }
+        if (fuelRemaining > 0 && machineTier == MachineTier.TIER_0 && machineActive == false) {
+            machineActive = true;
+            this.markForUpdate();
+        }
+        if (container.getStoredPower() > 0 && machineTier != MachineTier.TIER_0 && machineActive == false) {
+            if (inventory.getStackInSlot(1) != null) {
+                machineActive = true;
+                this.markForUpdate();
+
+            }
+        }
+
+
+        //for finding the fuel value
         if (fuelRemaining == 0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0 && machineTier == MachineTier.TIER_0) {
             if (inventory.getStackInSlot(0) != null || inventory.getStackInSlot(1) != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1066)